tools/monitoring/yaml-handler/parser.go

```
@@ -0,0 +1,116 @@
+limitations under the License.
+*/
+
+// The yamlHandler is responsible for fetching, parsing config yaml file. It also allows user to
+// retrieve a particular record from the yaml.
+
+package yamlHandler
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+type AlertConditions struct {
+	JobNameRegex string `yaml:"job-name-regex"`
+	Occurrences  int
+	JobsAffected string `yaml:"jobs-affected"`
+	PrsAffected  string `yaml:"prs-affected"`
+	Period       int
+}
+
+type PatternSpec struct {
+	ErrorPattern string `yaml:"error-pattern"`
+	Hint         string
+	Alerts       []AlertConditions
+}
+
+type YamlFile struct {
+	Spec []PatternSpec `yaml:"spec"`
+}
+
+type Output struct {
+	Hint         string
+	Occurrences  int
+	JobsAffected string
+	PrsAffected  string
+	Period       int
+}
+
+//GetSpec gets the spec for a particular error pattern and a matching job name pattern
+func GetSpec(f YamlFile, pattern, jobName string) (output Output, noMatchError error) {
+	noMatchError = errors.New(fmt.Sprintf("No spec found for pattern[%s] and jobName[%s]", pattern, jobName))
+	for _, patternSpec := range f.Spec {
+		if pattern == patternSpec.ErrorPattern {
+			noMatchError = errors.New(fmt.Sprintf("Spec found for pattern[%s], but no match for job name[%s]", pattern, jobName))
+			output.Hint = patternSpec.Hint
+			for _, alertCondition := range patternSpec.Alerts {
+				matched, err := regexp.MatchString(alertCondition.JobNameRegex, jobName)
+				if err != nil {
+					continue
+				}
+				if matched {
+					noMatchError = nil
+					output.JobsAffected = alertCondition.JobsAffected
+					output.Occurrences = alertCondition.Occurrences
+					output.PrsAffected = alertCondition.PrsAffected
+					output.Period = alertCondition.Period
+					break
+				}
+			}
+			break
+		}
+	}
+	return output, noMatchError
+}
+
+func getFileBytes(url string) []byte {
```

return an error here instead of logging it